jboss/container/wildfly/galleon/fp-content/java/added/src/main/resources/packages/wildfly.s2i.java/content/bin/java-conf.conf

```diff
@@ -5,6 +5,14 @@ export GC_MAX_METASPACE_SIZE=${GC_MAX_METASPACE_SIZE:-256}
 
 JAVA_OPTS="$(adjust_java_options ${JAVA_OPTS})"
 
+# CLOUD-3188 if JAVA_DIAGNOSTICS and the JDK specific diagnostics contain any GC log configuration, move the settings to PREPEND_JAVA_OPTS
+# to bypass the specific EAP checks done on JAVA_OPTS in standalone.sh that could remove the GC EAP specific log configurations
+JDK_DIAGNOSTICS=$(jvm_specific_diagnostics)
+if [ "x$JAVA_DIAGNOSTICS" != "x" ] && [ "x{JDK_DIAGNOSTICS}" != "x" ] && grep -q "\-Xlog:gc" <<< "${JDK_DIAGNOSTICS}"; then
```

Comment:
Actually, the grep part could fail in case JDK_DIAGNOSTICS change the syntax. Shouldn't we move the JDK_DIAGNOSTICS in all cases to PREPEND_JAVA_OPTS?

Reply:
I did in that way to apply the changes only when it is strictly necessary based on the current standalone.sh, and hence to minimize any possible side effects when the JDK_DIAGNOSTICS does not contain a -Xlog:gc setting.

However, I agree @jfdenise. Indeed, we should end up without duplicates in the JAVA_OPTS so any bits adding the same settings after processing this, does not make sense, so order here is irrelevant. Well, the -Xlog settings are a bit more tricky in this aspect since the same setting is overwritten by the following -Xlog argument configuring the same aspect. e.g if JDK_DIAGNOSTICS defines a log file by using -Xlog:gc:file=/one/path and the standalone.sh adds a different path using -Xlog:gc:file=/other/path, the latest configuration wins. It could make sense in this context, although enabling JAVA_DIAGNOSTICS is a user choice, so I could expect to see how those settings overrides the final JAVA_OPTS.

Reply:
I've just removed the grep check and appending the jvm specific diagnostics if it was activated and there is one